components/profile-controller/pkg/controller/profile/profile_controller.go

```diff
@@ -238,7 +240,7 @@ func (r *ReconcileProfile) Reconcile(request reconcile.Request) (reconcile.Resul
 		RoleRef: rbacv1.RoleRef{
 			APIGroup: "rbac.authorization.k8s.io",
 			Kind:     "ClusterRole",
-			Name:     ADMIN,
+			Name:     KUBEFLOW_PREFIX + ADMIN,
```

Comment:
I think it's preferable to be explicit here and pass the names we want to see created (eg kubeflow-editor instead of edit):
https://github.com/kubeflow/kubeflow/blob/5ebad34ce3e18727dda8288880cc25b3d230b568/components/profile-controller/pkg/controller/profile/profile_controller.go#L216

Reply:
@yanniszark 
> Regarding the need to patch the default ClusterRoles, isn't it only needed for ClusterRole view?
> We can get the permissions of the other ClusterRoles by adding their aggregationRules labelSelector.

IIUC, does this mean that we aggregate to default ClusterRoles Kubernetes provides and create RoleBindings against it? We were looking to aggregate the default ClusterRoles into our own Kubeflow ones, allowing for custom roles to be composed in the future. 

> 
> A final thought is this:
> Is it a stretch to define our own K8s kubeflow-viewer ClusterRole, with a set of permissions that is fairly stable? (eg Pods, Services, {Stateful, Daemon, Replica}Set, Deployments, Jobs, etc).

It shouldn't be a stretch, though the thinking here was to avoid having to maintain and update our own when Kubernetes provides us with its own default roles.

Reply:
I agree explicit naming would be better.

Reply:
> > Regarding the need to patch the default ClusterRoles, isn't it only needed for ClusterRole view?
> > We can get the permissions of the other ClusterRoles by adding their aggregationRules labelSelector.

> IIUC, does this mean that we aggregate to default ClusterRoles Kubernetes provides and create RoleBindings against it?

I had a lapse in my thinking there.
I was thinking of something along the lines of:
* We want kubeflow-viewer to get view permissions.
* Add aggregate-to-edit labelSelector to kubeflow-viewer, so that it will get the view permissions, like the edit ClusterRole does.

However, this is not ideal, as other applications in the Cluster may be aggregating to the edit ClusterRole.

To sum up, I see the following solutions:
* Edit the default ClusterRoles and add the necessary labels.
* Define our own base ClusterRoles.

Reply:
@krishnadurai @yanniszark so to summarize the key question is

1) Are Kubeflow roles the result of aggregating default K8s roles
or 
2) Do we define our roles for basic K8s permissions that users should have? i.e. we create kubeflow specific default roles that K8s defines and then we aggregate those roles

Option 1 has the advantages

* Depend on K8s to automatically update the roles with new resources over time

Option 1 has the disadvantage

* Kubeflow needs to modify the default roles to add the aggregation labels
* Kubeflow's default permissions are less explicit over time and potentially different K8s flavors/versions

Option 2 has the advantage 

* Kubeflow doesn't need to modify default cluster roles
* Kubeflow is very explicit about what the default permissions are

Option 2 has the disadvantage

* Kubeflow needs to explicitly modify the default roles over time
* Will we need to parameterize the default role because different K8s versions have different versions of resources? e.g. extension/v1beta1 vs. v1?

I tend to lean towards option 2 because it seems like if we start with 2 we could change our mind and still provide backwards compatiblity easier then with option 1.

If users want need to include the default rules in the aggregation then they could always manually add the label to the default roles. We could use that to validate the utility of doing that and eventually do it automatically.

Reply:
@jlewi This sounds good. Let's define our own Kubeflow roles for K8s resources. I'll change this PR to remove the labeling part and retain the Profile Controller and Kfam  parts. Let's have all ClusterRoles aggregate to Kubeflow roles in the manifest PR. I'll raise a separate PR defining a ClusterRole for Kubernetes resources.

Reply:
Another question is that should we start explicitly supporting a range of Kubernetes versions for each release if we are already not doing it?

Reply:
@krishnadurai LGTM to the first part.

@krishnadurai Yes I think we will need to start thinking about which versions of K8s we support and how to support multiple versions.

According to
https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles

It looks like role aggregation was introduced in 1.9. I think K8s is currently on 1.16 so that's 7 releases almost two years ago. 

I don't think we ever supported versions that low so I don't think this change is impacting our ability to support K8s versions currently used but let me know if I'm missing something.

Reply:
@jlewi I just wanted to ensure that the roles which we are defining would work across these supported K8s versions.